app/src/components/buttons/filterButton/filterContent/filterContent.jsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2024 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import classNames from 'classnames/bind';
+import { Button } from '@reportportal/ui-kit';
+import { useIntl } from 'react-intl';
+import { useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import isEqual from 'fast-deep-equal';
+import { COMMON_LOCALE_KEYS } from 'common/constants/localization';
+import { FilterInput } from './filterInput';
+import { messages } from './messages';
+import styles from './filterContent.scss';
+
+const cx = classNames.bind(styles);
+
+export const FilterContent = ({
+  setIsOpen,
+  setAppliedFiltersCount,
+  onFilterChange,
+  defaultFilters,
+  filters,
+  setFilters,
+  initialFilters,
+  filteredAction,
+}) => {
+  const { formatMessage } = useIntl();
+  const dispatch = useDispatch();
+
+  const closePopover = () => {
+    setIsOpen(false);
+    setFilters(initialFilters);
+  };
+
+  const clearAllFilters = () => {
+    setFilters(defaultFilters);
+  };
+
+  const handleApply = () => {
+    let appliedFiltersCount = 0;
+
+    const fields = Object.values(filters).reduce((acc, { filterName, value, condition }) => {
+      acc[filterName] = {
+        value,
+        condition,
+      };
+      appliedFiltersCount += value ? 1 : 0;
+
+      return acc;
+    }, {});
+    onFilterChange(fields);
+    setAppliedFiltersCount(appliedFiltersCount);
+    dispatch(filteredAction());
+    setIsOpen(false);
+  };
+
+  const isDefaultFilters = isEqual(defaultFilters, filters);
+  const isDefinedFilters = isEqual(initialFilters, filters);
```

Comment:
Do we really need to deeply compare these objects on every render?

Reply:
Done.